dix/req.c

```diff
@@ -2187,16 +2198,28 @@ static int dix_cas_rops_alloc(struct m0_dix_req *req)
 	struct m0_dix_rec_op       *rec_op;
 	uint32_t                    i;
 	uint32_t                    j;
+	uint32_t                    pa_idx;
+	uint32_t                    pa_nr;
 	uint32_t                    max_failures;
 	struct m0_dix_cas_rop     **map = rop->dg_target_rop;
 	struct m0_dix_cas_rop      *cas_rop;
 	struct m0_dix_pg_unit      *unit;
 	bool                        del_lock;
+	uint32_t                  **skipped_sdevs = NULL;
```

Comment:
Because I wanted to implement an elements position-dependent array with holes, hole is identified by NULL.
To use `uint32_t *skipped_sdevs` I need some hole identifier like 0xffffffff.
This question has already appeared twice, so I can change to `uint32_t *skipped_sdevs` with hole identifier if usage of double pointer has bad readability. @nikitadanilov what do you think?

Reply:
Rewritten to uint32_t* as @huanghua78 suggested.